From 50197c9e56fdbbb831bd35f577f1409a8cd35ccd Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Tue, 18 Jan 2022 15:27:46 +0100
Subject: [PATCH] typeC: Add kABI placeholders

References: bsn#129 bsc#1183030
Patch-mainline: never, kABI
Modified-by-SEL: Yes, modified due to different context

Adding it to the most important structures

Signed-off-by: Oliver Neukum <oneukum@suse.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/usb/typec/bus.h   |    2 ++
 drivers/usb/typec/class.c |    4 ++++
 2 files changed, 6 insertions(+)

--- a/drivers/usb/typec/bus.h
+++ b/drivers/usb/typec/bus.h
@@ -22,6 +22,8 @@ struct altmode {
 
 	struct altmode			*partner;
 	struct altmode			*plug[2];
+
+	void *				suse_kabi_padding;
 };
 
 #define to_altmode(d) container_of(d, struct altmode, adev)
--- a/drivers/usb/typec/class.c
+++ b/drivers/usb/typec/class.c
@@ -33,6 +33,8 @@ struct typec_partner {
 	struct usb_pd_identity		*identity;
 	enum typec_accessory		accessory;
 	struct ida			mode_ids;
+
+	void				*suse_kabi_padding;
 };
 
 struct typec_port {
@@ -54,6 +56,8 @@ struct typec_port {
 
 	const struct typec_capability	*cap;
 	const struct typec_operations   *ops;
+
+	void				*suse_kabi_padding;
 };
 
 #define to_typec_port(_dev_) container_of(_dev_, struct typec_port, dev)
